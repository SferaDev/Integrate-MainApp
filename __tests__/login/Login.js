import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import LogIn from '../../components/login/login';

const navigation = {navigate: jest.fn()};
let wrapper;
let instance;

describe('Test group for EntityList', function () {
    beforeAll(() => {
        jest.useFakeTimers();
        jest.mock('react-native-maps', () => require.requireActual('../../__mocks__/react-native-maps').default);
        configure({adapter: new Adapter()});
    });

    beforeEach(function () {
        // Before each: Shallows the EntityList component
        wrapper = shallow(<LogIn navigation={navigation}/>);
        instance = wrapper.instance();
    });

    afterEach(function () {
        // After each: Clears the wrapper
        wrapper = null;
        instance = null;
    });

    test('renders Login correctly', () => {

        const tree = renderer.create(<LogIn/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('login() is callable and returns nothing', () => {
        expect(instance.login()).toBe(undefined);
    });

    it('autologin() when token is null is callable and returns nothing', () => {
        expect(instance.autologin(null)).toBe(undefined);
    });

    it('autologin() when token is not null is callable and returns nothing', () => {
        expect(instance.autologin("MI_TOKEN")).toBe(undefined);
    });

    it('updateNifNie() is callable and returns nothing', () => {
        expect(instance.updateNifNie('')).toBe(undefined);
    });

    it('updatePassword() is callable and returns nothing', () => {
        expect(instance.updatePassword('')).toBe(undefined);
    });

    it('showError() is callable and returns nothing', () => {
        expect(instance.showError()).toBe(undefined);
    });

    it('updateError() is callable and returns nothing', () => {
        expect(instance.updateError()).toBe(undefined);
    });

    describe("isEmpty() tests", () => {
        it('isEmpty() when NifNie and Password are empty then returns true', () => {
            instance.state.nifnie = '';
            instance.state.password = '';
            expect(instance.isEmpty()).toBe(true);
        });

        it('isEmpty() when NifNie is empty and Password is filled then returns true', () => {
            instance.state.nifnie = '';
            instance.state.password = 'QWERTY1234';
            expect(instance.isEmpty()).toBe(true);
        });

        it('isEmpty() when Password is empty and NifNie is empty then returns true', () => {
            instance.state.nifnie = '123456789';
            instance.state.password = '';
            expect(instance.isEmpty()).toBe(true);
        });

        it('isEmpty() when NifNie and Password are filled then returns false', () => {
            instance.state.nifnie = '123456789';
            instance.state.password = 'QWERTY1234';
            expect(instance.isEmpty()).toBe(false);
        });
    });

    it('moveUp() is callable and returns nothing', () => {
        expect(instance.moveUp()).toBe(undefined);
    });

    it('moveDown() is callable and returns nothing', () => {
        expect(instance.moveDown()).toBe(undefined);
    });

    it('componentWillUnmount() is callable and returns nothing', () => {
        expect(instance.componentWillUnmount()).toBe(undefined);
    });

    it('restorePassword() is callable and returns nothing', () => {
        expect(instance.restorePassword()).toBe(undefined);
    });

    it('navigateHome() is callable and returns nothing', () => {
        expect(instance.navigateHome()).toBe(undefined);
    });

    describe("getButtonBackground() and getButtonColor() tests", () => {

        it('getButtonBackground() is callable and returns a color code', () => {
            expect(typeof instance.getButtonBackground()).toBe("string");
        });

        it('getButtonBackground() is callable and returns a color code', () => {
            instance.state.nifnie = '123456789';
            instance.state.password = 'QWERTY1234';
            expect(typeof instance.getButtonBackground()).toBe("string");
        });

        it('getButtonColor() is callable and returns a color code', () => {
            expect(typeof instance.getButtonColor()).toBe("string");
        });

        it('getButtonColor() is callable and returns a color code', () => {
            instance.state.nifnie = '123456789';
            instance.state.password = 'QWERTY1234';
            expect(typeof instance.getButtonColor()).toBe("string");
        });
    });
});
