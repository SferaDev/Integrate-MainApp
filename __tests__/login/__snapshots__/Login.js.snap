// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test group for EntityList renders Login correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "height": "100%",
        "paddingTop": 63,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../Images/bg.jpg",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "100%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../Images/ic_launcher.png",
          }
        }
        style={
          Object {
            "height": 200,
            "width": 200,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../Images/integrateHeader1.png",
          }
        }
        style={
          Array [
            Object {
              "height": 58,
              "width": 293,
            },
            Object {
              "marginBottom": 20,
              "marginTop": 10,
            },
          ]
        }
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Introduir NIF/NIE"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255,255,255,0.85)",
              "borderColor": "#0c59cf",
              "borderRadius": 2,
              "borderWidth": 1,
              "height": 35,
              "justifyContent": "center",
              "margin": 10,
              "padding": 0,
              "paddingLeft": 5,
              "width": 260,
            },
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Introduir contrasenya"
        secureTextEntry={true}
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255,255,255,0.85)",
              "borderColor": "#0c59cf",
              "borderRadius": 2,
              "borderWidth": 1,
              "height": 35,
              "justifyContent": "center",
              "margin": 10,
              "padding": 0,
              "paddingLeft": 5,
              "width": 260,
            },
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
        value=""
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "textDecorationLine": "underline",
          }
        }
      >
        He oblidat la contrasenya?
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "borderColor": "#0c59cf",
                "borderRadius": 4,
                "borderWidth": 1,
                "height": 30,
                "justifyContent": "center",
                "margin": 10,
                "width": 260,
              },
              Object {
                "backgroundColor": "#CCC",
              },
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#666",
              "fontWeight": "bold",
            }
          }
        >
          Entra
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgba(0,0,0,0.40)",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "padding": 20,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "display": "none",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          El Nie / Nif o la contrasenya s√≥n incorrectes
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignSelf": "center",
                "backgroundColor": "#67ACB1",
                "borderColor": "#094671",
                "borderRadius": 4,
                "borderWidth": 3,
                "display": "flex",
                "flexDirection": "column",
                "height": 40,
                "justifyContent": "center",
                "margin": 10,
                "width": 100,
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#094671",
                  "fontWeight": "bold",
                },
              ]
            }
          >
            CLOSE
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
